pkg/image/registry/imagestreamimport/rest.go

```diff
@@ -147,12 +148,20 @@ func (r *REST) Create(ctx apirequest.Context, obj runtime.Object, _ bool) (runti
 	}
 
 	// only load secrets if we need them
-	credentials := importer.NewLazyCredentialsForSecrets(func() ([]kapi.Secret, error) {
+	credentials := importer.NewLazyCredentialsForSecrets(func() ([]kapiv1.Secret, error) {
 		secrets, err := r.secrets.ImageStreamSecrets(namespace).Secrets(isi.Name, metav1.ListOptions{})
 		if err != nil {
 			return nil, err
 		}
-		return secrets.Items, nil
+		secretsv1 := make([]kapiv1.Secret, len(secrets.Items))
```

Comment:
@soltysh this might be interesting for you... this is needed to allow registry use the credentials (registry use external client). when we switch to external client in image controller, this can be nuked.